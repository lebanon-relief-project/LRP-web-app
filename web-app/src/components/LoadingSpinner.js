import React from "react";
import styled from "styled-components";
import loadingSpinnerGif from "../assets/gifs/LoadingSpinner.gif";

const LoadingSpinner = () => {
  return <StyledLoadingSpinner src={loadingSpinnerGif} alt="loading..." />;
};

const StyledLoadingSpinner = styled.img`
  heigth: auto;
  width: 10%;
`;

export default LoadingSpinner;
