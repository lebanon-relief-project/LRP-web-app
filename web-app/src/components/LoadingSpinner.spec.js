import React from "react";
import { render } from "@testing-library/react";
import LoadingSpinner from "./LoadingSpinner";
import "jest-styled-components";

describe("the LoadingSpinner component", () => {
  let container;
  it("should match the snapshot", () => {
    ({ container } = renderLoadingSpinner());

    expect(container).toMatchSnapshot();
  });
});

const renderLoadingSpinner = () => {
  return render(<LoadingSpinner />);
};
