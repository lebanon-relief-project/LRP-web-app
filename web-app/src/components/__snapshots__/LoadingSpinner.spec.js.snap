// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the LoadingSpinner component should match the snapshot 1`] = `
.c0 {
  heigth: auto;
  width: 10%;
}

<div>
  <img
    alt="loading..."
    class="c0"
    src="LoadingSpinner.gif"
  />
</div>
`;
